src/ucp/api/ucp.h

```diff
@@ -13,6 +13,10 @@
 #include <ucs/config/types.h>
 #include <ucs/sys/math.h>
 #include <stdio.h>
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
```

Comment:
this is a really bad idea. We have to figure out how to handle this. I don't want to API level dependance on HWLOC either. This pretty much leaves us only with the option of defining own interface for cpu query ...